activation/handler_v2_test.go

```diff
@@ -83,9 +95,35 @@ func (h *handlerMocks) expectVerifyNIPoST(atx *wire.ActivationTxV2) {
 	).Return(poetLeaves, nil)
 }
 
+func (h *handlerMocks) expectVerifyNIPoSTs(
+	atx *wire.ActivationTxV2,
+	equivocationSet []types.NodeID,
+	poetLeaves []uint64,
+) {
+	for i, nipost := range atx.NiPosts {
+		for _, post := range nipost.Posts {
+			h.mValidator.EXPECT().PostV2(
+				gomock.Any(),
+				equivocationSet[post.MarriageIndex],
+				gomock.Any(),
+				wire.PostFromWireV1(&post.Post),
+				nipost.Challenge.Bytes(),
+				post.NumUnits,
+				gomock.Any(),
+			)
+		}
+		h.mValidator.EXPECT().PoetMembership(
+			gomock.Any(),
+			gomock.Any(),
+			nipost.Challenge,
+			gomock.Any(),
+		).Return(poetLeaves[i], nil)
+	}
+}
+
 func (h *handlerMocks) expectStoreAtxV2(atx *wire.ActivationTxV2) {
-	h.mbeacon.EXPECT().OnAtx(gomock.Any())
-	h.mtortoise.EXPECT().OnAtx(gomock.Any(), gomock.Any(), gomock.Any())
+	h.mbeacon.EXPECT().OnAtx(fixture.MatchId(atx.ID()))
```

Comment:
This achieves the same effect without the additional type:

```suggestion
	h.mbeacon.EXPECT().OnAtx(h.mbeacon.EXPECT().OnAtx(gomock.Cond(func(x any) bool {
		return x.(*types.ActivationTx).ID() == atx.ID()
	})))
```

I even contributed to go-mock to hopefully not need the cast any more by making `go.Cond` generic: https://github.com/uber-go/mock/pull/183

Reply:
I added the helper to avoid doing this construct exactly.

Reply:
but why? what if it would already be generic?

```go
h.mbeacon.EXPECT().OnAtx(gomock.Cond(func(x *types.ActivationTx) bool { return x.ID() == atx.ID() })))
```

looks to me straight forward to what it does, is just as short and doesn't require a new type in a different package?

Reply:
But it's not generic yet :shrug:

Reply:
Removed the Matcher